mirage-xen does not work with OCaml 4.05

Linking the noop example from mirage-skeleton gives:

```
$ make
mirage build
_build/main.native.o: In function `camlBigarray__map_file_inner_1994':
/home/user/.opam/4.05.0/build/ocaml/otherlibs/bigarray/bigarray.ml:137: undefined reference to `caml_ba_map_file'
_build/main.native.o: In function `camlBigarray__49':
:(.data+0x67c70): undefined reference to `caml_ba_map_file'
```

--- a/xen-ocaml/build.sh
+++ b/xen-ocaml/build.sh
@@ -50,8 +50,8 @@ case `ocamlopt -version` in
 4.05.*)
   echo Applying OCaml 4.05 config
   cp config/version-405.h ocaml-src/byterun/caml/version.h
-  BIGARRAY_OBJ=""
-  CFLAGS="-D__ANDROID__"
+  BIGARRAY_OBJ="mmap_unix.o"
+  CFLAGS="-D__ANDROID__ $CFLAGS"
   ;;
 *)
   echo unsupported OCaml version `ocamlopt -version`

